samza-core/src/main/java/org/apache/samza/storage/StoreBackupEngine.java

```diff
@@ -0,0 +1,185 @@
+package org.apache.samza.storage;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.Config;
+import org.apache.samza.metrics.MetricsRegistryMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.conf.*;
+
+// Class to hold common methods and variables for backup storage and restore process
+public class StoreBackupEngine {
+  protected static final Logger LOGGER = LoggerFactory.getLogger(StoreBackupEngine.class);
+  protected FileSystem fs;
+  protected Path hdfsPath;
+  protected Path dbPath;
+  public StoreBackupMetrics metrics;
+
+  public StoreBackupEngine(Config storeConfig) {
+    metrics = new StoreBackupMetrics("backup", new MetricsRegistryMap("registrymap"));
+
+    // Setup fs configuration settings
+    Configuration conf = new Configuration();
+    // If core site path file is specified
+    if (storeConfig.containsKey("hdfs.coreSitePath")){
+      Path coreSitePath = new Path(storeConfig.get("hdfs.coreSitePath"));
+      conf.addResource(coreSitePath);
+    } else {
+      // Otherwise, set a default localhost port for fs.defaultFS - hdfs://localhost:9000
+      conf.set("fs.default.name", "hdfs://localhost:9000");
+    }
+
+    // Initialize path variables
+    // Method: config.get(key, defaultVal)
+    dbPath = new Path(storeConfig.get("rocksdb.dbDir", System.getProperty("java.io.tmpdir") + "/DB"));
+
+    // Set where to store backup files on hdfs
+    // TODO: change default value to something else for backup since obviously below default only works for me
+    hdfsPath = new Path(storeConfig.get("hdfs.backupDir", "/User/eripan/backupTest/"));
+
+    // Possible IOException when using filesystem api
+    try {
+      fs = FileSystem.get(conf);
+
+      // Create directory for hdfs backup location
+      // No need to check if localPath is dir since it's checked when rocksdb is created
+      if (fs.exists(hdfsPath)) {
+        if (fs.isFile(hdfsPath)) {
+          throw new SamzaException("Error creating backup in hdfs: Path is already a file");
+        }
+      } else {
+        fs.mkdirs(hdfsPath);
+      }
+
+    } catch(IOException e) {
+      throw new SamzaException("Failed to create hdfs filesystem: " + e);
```

Comment:
For exceptions, use the constructor that takes the cause as an argument instead of adding its toString representation (e.getMessage()) to this exception message.